lib/charms/tls_certificates_interface/v4/tls_certificates.py

```diff
@@ -1105,7 +1105,7 @@ def _get_app_or_unit(self) -> Union[Application, Unit]:
         raise TLSCertificatesError("Invalid mode")
 
     @property
-    def private_key(self) -> PrivateKey | None:
+    def private_key(self) -> Union[PrivateKey, None]:
```

Comment:
In the case of a Union with None, I think using `Optional` is more readable and will also solve the problem.

Applies to changes below as well.

Reply:
Done